### PR TITLE
refactor(router): domain and diesel model changes for merchant_connector_account create v2 flow

### PR DESCRIPTION
**Pull request by sai-harsha-vardhan**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

domain and diesel model changes for merchant_connector_account create v2 flow

### Additional Changes

- [ ] This PR modifies the API contract
- [x] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

Tested manually
Sanity MCA Create and subsequent flows like payment / capture / refund with v1 version
Sanity Sessions flow
Sanity Incoming webhooks flow
Sanity MCA create / update / retrieve flows

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: crates/diesel_models/src/merchant_connector_account.rs [hunk 1]
```diff
@@ -47,6 +58,65 @@ pub struct MerchantConnectorAccount {
...
+    router_derive::DebugAsDisplay,
+)]
+#[diesel(table_name = merchant_connector_account, check_for_backend(diesel::pg::Pg))]
+pub struct MerchantConnectorAccount {
+    pub merchant_id: id_type::MerchantId,
+    pub connector_name: String,
+    pub connector_account_details: Encryption,
+    pub disabled: Option<bool>,
+    #[diesel(deserialize_as = super::OptionalDieselArray<serde_json::Value>)]
+    pub payment_methods_enabled: Option<Vec<serde_json::Value>>,
```
Comment (jarnura) [inline crates/diesel_models/src/merchant_connector_account.rs:89]:
```suggestion
    pub payment_methods_enabled: Option<Vec<pii::SecretSerdeValue>>,
```

File: crates/diesel_models/src/merchant_connector_account.rs [hunk 2]
```diff
@@ -47,6 +58,65 @@ pub struct MerchantConnectorAccount {
...
+    pub connector_type: storage_enums::ConnectorType,
+    pub metadata: Option<pii::SecretSerdeValue>,
+    pub connector_label: Option<String>,
+    pub frm_configs: Option<Secret<serde_json::Value>>,
```
Comment (jarnura) [inline crates/diesel_models/src/merchant_connector_account.rs:93]:
```suggestion
    pub frm_configs: Option<Secret<pii::SecretSerdeValue>>,
```

File: crates/diesel_models/src/merchant_connector_account.rs [hunk 3]
```diff
@@ -47,6 +58,65 @@ pub struct MerchantConnectorAccount {
...
+    pub created_at: time::PrimitiveDateTime,
+    pub modified_at: time::PrimitiveDateTime,
+    pub connector_webhook_details: Option<pii::SecretSerdeValue>,
+    #[diesel(deserialize_as = super::OptionalDieselArray<pii::SecretSerdeValue>)]
+    pub frm_config: Option<Vec<Secret<serde_json::Value>>>,
```
Comment (jarnura) [inline crates/diesel_models/src/merchant_connector_account.rs:98]:
```suggestion
    pub frm_config: Option<Vec<Secret<pii::SecretSerdeValue>>>,
```

Comment (jarnura) [reply crates/diesel_models/src/merchant_connector_account.rs:98]: There are two frm configs we should remove one which is deprecated.

File: crates/diesel_models/src/merchant_connector_account.rs [hunk 4]
```diff
@@ -47,6 +58,65 @@ pub struct MerchantConnectorAccount {
...
+    Debug,
+    serde::Serialize,
+    serde::Deserialize,
+    Identifiable,
+    Queryable,
+    Selectable,
+    router_derive::DebugAsDisplay,
+)]
+#[diesel(table_name = merchant_connector_account, check_for_backend(diesel::pg::Pg))]
+pub struct MerchantConnectorAccount {
```
Comment (jarnura) [inline crates/diesel_models/src/merchant_connector_account.rs:83]: connector_id is missing

**Review by jarnura — CHANGES_REQUESTED** [review]

File: crates/diesel_models/src/merchant_connector_account.rs (hunk 3 above)
Comment (jarnura) [reply crates/diesel_models/src/merchant_connector_account.rs:98]: Address all these in other related types too (like `MerchantConnectorAccountNew`, update types, update internal types)

**Review by jarnura — CHANGES_REQUESTED** [review]

File: crates/diesel_models/src/query/merchant_connector_account.rs [hunk 5]
```diff
@@ -149,3 +159,122 @@ impl MerchantConnectorAccount {
...
+        generics::generic_find_one::<<Self as HasTable>::Table, _, _>(
+            conn,
+            dsl::merchant_id
+                .eq(merchant_id.to_owned())
+                .and(dsl::connector_label.eq(connector_label.to_owned())),
+        )
+        .await
+    }
+
+    pub async fn find_by_profile_id_connector_name(
```
Comment (jarnura) [inline crates/diesel_models/src/query/merchant_connector_account.rs:204]: we can deprecated this function and its usage, this related functionalities should be handled by id

File: crates/diesel_models/src/query/merchant_connector_account.rs [hunk 6]
```diff
@@ -149,3 +159,122 @@ impl MerchantConnectorAccount {
...
+        generics::generic_find_one::<<Self as HasTable>::Table, _, _>(
+            conn,
+            dsl::profile_id
+                .eq(profile_id.to_owned())
+                .and(dsl::connector_name.eq(connector_name.to_owned())),
+        )
+        .await
+    }
+
+    pub async fn find_by_merchant_id_connector_name(
```
Comment (jarnura) [inline crates/diesel_models/src/query/merchant_connector_account.rs:218]: we can deprecated this function and its usage, this related functionalities should be handled by id

**Review by jarnura — CHANGES_REQUESTED** [review]

File: crates/router/src/core/admin.rs [hunk 7]
```diff
@@ -2673,6 +2697,45 @@ pub async fn update_payment_connector(
         )
         .await?,
     };
+    #[cfg(all(feature = "v2", feature = "merchant_connector_account_v2"))]
+    let payment_connector = storage::MerchantConnectorAccountUpdate::Update {
+        merchant_id: None,
```
Comment (jarnura) [inline crates/router/src/core/admin.rs:2702]: This field should be deprecated, Merchant connector accounts merchant id we should not update.

File: crates/router/src/core/admin.rs [hunk 8]
```diff
@@ -2673,6 +2697,45 @@ pub async fn update_payment_connector(
...
+        connector_type: Some(req.connector_type),
+        connector_name: None,
```
Comment (jarnura) [inline crates/router/src/core/admin.rs:2704]: This field should be deprecated, Merchant connector accounts connector name we should not update.

**Review by jarnura — CHANGES_REQUESTED** [review]

File: crates/diesel_models/src/query/merchant_connector_account.rs [hunk 9]
```diff
@@ -149,3 +159,86 @@ impl MerchantConnectorAccount {
...
+            result => result,
+        }
+    }
+
+    pub async fn delete_by_id(conn: &PgPooledConn, id: &str) -> StorageResult<bool> {
+        generics::generic_delete::<<Self as HasTable>::Table, _>(conn, dsl::id.eq(id.to_owned()))
+            .await
+    }
+
+    pub async fn find_by_merchant_id_connector(
```
Comment (jarnura) [inline crates/diesel_models/src/query/merchant_connector_account.rs:190]: This function also we can deprecate.

**Review by jarnura — CHANGES_REQUESTED** [review]

**Review by jarnura — APPROVED** [review]

**Review by prajjwalkumar17 — DISMISSED** [review]

**Review by ThisIsMani — APPROVED** [review]
Dashboard specific changes looks fine.

**Review by apoorvdixit88 — APPROVED** [review]

**Review by jarnura — APPROVED** [review]

**Review by prajjwalkumar17 — APPROVED** [review]

**Review by apoorvdixit88 — APPROVED** [review]

**Pull request merged**